ui/package.json

```diff
@@ -16,8 +16,8 @@
     "test:watch": "jest --watch",
     "test:debug": "node --inspect-brk node_modules/.bin/jest --runInBand --watch",
     "lint": "yarn run eslint; yarn run tsc",
-    "test:e2e": "CYPRESS_baseUrl=http://localhost:8888 cypress run --browser chrome --headless --reporter junit --reporter-options 'mochaFile=cypress/results/results-[hash].xml'",
-    "test:e2e:headed": "CYPRESS_baseUrl=http://localhost:8888 cypress open --browser chrome",
+    "test:e2e": "(nohup yarn --cwd $(git rev-parse --show-toplevel)/etc/oauth2-server-mock start > oauth2-mock.log 2>&1 &) && CYPRESS_oauth2ServerURL=http://${OAUTH2_HOSTNAME}:${OAUTH2_PORT} cypress run --browser chrome --headless --reporter junit --reporter-options 'mochaFile=cypress/results/results-[hash].xml'",
```

Comment:
Shouldn't we rather create a new `test:oauth-mock` target that simply starts the mock server and reuse it with both `test:e2e` and `test:e2e:headed`? The targets should not expect extra settings of env variables, or do they? ... can't we use `${OAUTH2_PORT:-8087}`

Reply:
Environmental variables are set by executing `oauth-for-chronograf.sh` in the chronograf/etc/oauth2-sever-mock directory, that will also execute `env.sh` in the said directory, which means  `${OAUTH2_PORT}` will always be set. So may I ask why `${OAUTH2_PORT:-8087}`? In that case, shouldn't there be also `${OAUTH2_HOSTNAME:-localhost}`, if the environmental variable isn't set? Both variables are passed this way to ensure that they are dynamic. If they were set in the cypress.json and the user changed the OAuth2 Server mock hostname, port, or both, the tests would fail.

Reply:
Added `test:oauth-mock` to start the mock server and used it in `test:e2e` and `test:e2e:headed`. Adjusted the parameter expansion according to your recommendation.